xonsh/aliases.py

```diff
@@ -127,6 +124,29 @@ def source_alias(args, stdin=None):
             execx(fp.read(), 'exec', builtins.__xonsh_ctx__)
 
 
+def source_cmd(args, stdin=None):
+    """Simple cmd.exe-specific wrapper around source-foreign."""
+    args = list(args)
+    fpath = locate_binary(args[0])
+    args[0] = fpath if fpath else args[0]
+    args.insert(0, 'cmd')
+    args.append('--interactive=0')
+    args.append('--sourcer=call')
+    args.append('--envcmd=set')
+    args.append('--safe=0')
+    args.append('--use-tmpfile=1')
+    return source_foreign(args, stdin=stdin)
+
+
+DEFAULT_ALIASES.update({
+    'source-foreign': source_foreign,
+    'source-bash': ['source-foreign', 'bash', '--sourcer=source'],
+    'source-zsh': ['source-foreign', 'zsh', '--sourcer=source'],
+    'source-cmd': source_cmd,
+    'source-alias': source_alias,
+})
```

Comment:
I had already replaced all the source functions, when I figured I couldn't `source-cmd` unless it was a normal alias function.